src/ui/common/src/stories/DataListTable.stories.tsx

```diff
@@ -0,0 +1,232 @@
+      value: '$500',
+      status: ExecutionStatus.Succeeded,
+    },
+    {
+      metricId: '5',
+      name: 'more_metrics',
+      value: '$500',
+      status: ExecutionStatus.Succeeded,
+    },
+  ];
+
+  // TODO: Change this type to something more generic.
+  // Also make this change in WorkflowsTable, I think we can just use Data here if we add JSX.element to Data's union type.
+  const mockArtifactList: PaginatedSearchTableData = {
+    schema: {
+      fields: [
+        { name: 'name', type: 'varchar' },
+        { name: 'created_at', displayName: 'Created At', type: 'varchar' },
+        { name: 'workflow', type: 'varchar' },
+        { name: 'type', type: 'varchar' },
+        { name: 'metrics', type: 'varchar' },
+        { name: 'checks', type: 'varchar' },
+      ],
+      pandas_version: '1.5.1',
+    },
+    data: [
+      {
+        name: {
+          name: 'churn_model',
+          url: '/data',
+          status: ExecutionStatus.Succeeded,
+        },
+        created_at: '11/1/2022 2:00PM',
+        workflow: {
+          name: 'train_churn_model',
+          url: '/workflows',
+          status: ExecutionStatus.Running,
+        },
+        type: 'sklearn.linear, Linear Regression',
+        metrics: [],
+        checks: checkPreviews,
+      },
+      {
+        name: {
+          name: 'predict_churn_dataset',
+          url: '/workflows',
+          status: ExecutionStatus.Running,
+        },
+        created_at: '11/1/2022 2:00PM',
+        workflow: {
+          name: 'monthly_churn_prediction',
+          url: '/workflows',
+          status: ExecutionStatus.Succeeded,
+        },
+        type: 'pandas.DataFrame',
+        metrics: metricsShort,
+        checks: checkPreviews,
+      },
+      {
+        name: {
+          name: 'label_classifier',
+          url: '/data',
+          status: ExecutionStatus.Pending,
+        },
+        created_at: '11/1/2022 2:00PM',
+        workflow: {
+          name: 'label_classifier_workflow',
+          url: '/workflows',
+          status: ExecutionStatus.Registered,
+        },
+        type: 'parquet',
+        metrics: metricsShort,
+        checks: checkPreviews,
+      },
+    ],
+    meta: [
+      {
+        name: 'churn_model',
+        created_at: '11/1/2022 2:00PM',
+        workflow: 'train_churn_model',
+        type: 'sklearn.linear, Linear Regression',
+        metrics: metricsShort,
+        checks: checkPreviews,
+      },
+      {
+        name: 'predict_churn_dataset',
+        created_at: '11/1/2022 2:00PM',
+        workflow: 'monthly_churn_prediction',
+        type: 'pandas.DataFrame',
+        metrics: metricsShort,
+        checks: checkPreviews,
+      },
+      {
+        name: 'label_classifier',
+        created_at: '11/1/2022 2:00PM',
+        workflow: 'label_classifier_workflow',
+        type: 'parquet',
+        metrics: metricsShort,
+        checks: checkPreviews,
+      },
+    ],
+  };
+
+  const onGetColumnValue = (row, column) => {
```

Comment:
I recall this has appeared a few times, is there a way to deduplicate?

Reply:
The purpose of this function is to allow the data tables to render arbitrary react components in the table cell and still allow for things like sorting and searching.

I can update the documentation here to make this more clear.